presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -373,10 +375,12 @@ public LocalExecutionPlan plan(
                         maxPagePartitioningBufferSize));
     }
 
-    public LocalExecutionPlan plan(TaskContext taskContext,
+    public LocalExecutionPlan plan(
+            TaskContext taskContext,
             PlanNode plan,
             List<Symbol> outputLayout,
             Map<Symbol, Type> types,
+            List<PlanNodeId> partitionedSourceOrder,
```

Comment:
It is the order in which you must start partitioned splits to avoid deadlocks... so for a collocated join, you must start all build splits before you schedule any probes.